api/src/main/java/org/openmrs/api/db/hibernate/HibernateProgramWorkflowDAO.java

```diff
@@ -460,8 +460,8 @@ public void purgeProgramAttributeType(ProgramAttributeType type) {
                             " from patient p " +
                             " join patient_program pp on p.patient_id = pp.patient_id and p.patient_id in (" + commaSeperatedPatientIds + ")" +
                             " join patient_program_attribute ppa on pp.patient_program_id = ppa.patient_program_id and ppa.voided=0" +
-                            " join program_attribute_type ppt on ppa.attribute_type_id = ppt.program_attribute_type_id and ppt.name ='" + attributeName + "' "+
-                            " LEFT OUTER JOIN concept_name cn on ppa.value_reference = cn.concept_id and cn.concept_name_type= 'FULLY_SPECIFIED' and cn.voided=0 and ppt.datatype like '%ConceptDataType%'" +
+                            " join program_attribute_type ppt on ppa.attribute_type_id = ppt.program_attribute_type_id and ppt.name ='" + attributeName + "'"+
+                            " and ppt.datatype like '%ConceptDataType%' LEFT OUTER JOIN concept_name cn on ppa.value_reference = cn.concept_id and cn.concept_name_type= 'FULLY_SPECIFIED' and cn.voided=0" +
```

Comment:
@akshay-jha why are you making this new change ?

Reply:
I have changed the order of that clause because if you provide a program attribute type whose datatype is not equivalent to containing "ConceptDataType", then the query was breaking earlier and throwing a number format exception. When the datatype is "ConceptDataType" the value reference is supposed to contain the concept_id but then value reference is a text field, so this is to cover the corner case and making sure that exception is not thrown.

Reply:
My only worry is about other methods that were putting in use the query and had it work well,could there be any other place where the query is being used??

Reply:
It is being used in `ProgramWorkflowServiceImpl` but none of the other classes seem to be calling this method of the service Impl class, maybe that is the reason that this bug was never discovered.

Reply:
Let me know in case this needs to be reverted.

Reply:
Could we update this query to use parameters instead of string concatenation while we're doing this? So, `ppt.name ='" + attributeName + "'"` -> `ppt.name = :attributeName`

Reply:
Sure, have added those changes now @ibacher